src/parser/input_utils.rs

```diff
@@ -127,8 +113,55 @@ pub fn new_attr_set(attr_pairs: Vec<(GreenNode, GreenNode)>) -> GreenNode {
     GreenNode::new(attr_set_kind, tokens)
 }
 
+pub(crate) struct Key {
+    val: SmlStr,
+}
```

Comment:
I think a simple newtype struct would be a better fit, unless you intend to add more fields.
```suggestion
pub(crate) struct Key(SmlStr);
```